linter.py

```diff
@@ -41,15 +44,16 @@ class Codeclimate(Linter):
     error_stream = util.STREAM_BOTH
     word_re = None
 
+    root = None
+    path = None
+
     def cmd(self):
-        """Construct a cmd to provide a relative path to 'codeclimate analyze'."""
-        result = ['codeclimate', 'analyze', '-e', 'structure', '-e', 'duplication']
-        relative_file_name = None
-        for folder in self.view.window().folders():
-            if self.filename.find(folder) == 0:
-                relative_file_name = self.filename.replace(folder + '/', '')
-        if relative_file_name == None:
-            return result
-        result.append(relative_file_name)
-        persist.debug(result)
-        return result
+        """Set working directory and run 'codeclimate analyze'."""
+        if self.context.get('project_root') is None:
```

Comment:
Does this new code properly read the `.codeclimate.yml` file found at the project root?

Reply:
Yes, of course. 

The defaults are initially still set to `${project}` ([linter.py#L23](https://github.com/meengit/SublimeLinter-contrib-codeclimate/blob/42c2145a98a08759f64785568488a1ab824c9189/linter.py#L23)).

If the project has the type of None and no project is configured, I try to find the path of the first opened folder in the folder pane ([linter.py#L52-#L53](https://github.com/meengit/SublimeLinter-contrib-codeclimate/blob/42c2145a98a08759f64785568488a1ab824c9189/linter.py#L52-#L53)). If no folder is open in the folder pane, the folder of the currently open file taking place ([linter.py#L54-#L55](https://github.com/meengit/SublimeLinter-contrib-codeclimate/blob/42c2145a98a08759f64785568488a1ab824c9189/linter.py#L54-#L55)).

As far as I know, the Codeclimate CLI has no argument to set a path to a `.codeclimate.yml` file. Instead, it expects such one in the working directory of execution.

Reply:
I understand you did your changes for your personal/individual use. However, your changes helped me a lot. I do not "expect" that you will use my changes. Instead, I hope you can give it a try/test it. It would enable me to collect outside feedback to prevent mistakes I didn't/can't discover in my development environments.

Reply:
@schuylr 

Before I go to package control directly, I will try bring my changes back to the codeclimate repository in a pull request. Is it okey for you, when I mention you in the copyrights?

```python
# Updated by Schuyler Jager & Andreas for SublimeLinter 4
# Copyright (c) 2020 Schuyler Jager & Andreas
# License: MIT
```

[`linter.py#L8-#L10`](https://github.com/meengit/SublimeLinter-contrib-codeclimate/blob/7314e2d49b0df88d3fc0a5a00763065037817d5f/linter.py#L8-#L10)